clar2wasm/src/wasm_generator.rs

```diff
@@ -3048,19 +3068,56 @@ impl WasmGenerator {
         for operand in operands {
             self.traverse_expr(builder, operand)?;
         }
+
         self.visit_comparison(builder, expr, func, operands)
     }
 
-    fn traverse_logical(
+    fn traverse_comparison_equals(
+        &mut self,
+        builder: &mut InstrSeqBuilder,
+        _expr: &SymbolicExpression,
+        func: NativeFunctions,
+        operands: &[SymbolicExpression],
+    ) -> Result<(), GeneratorError> {
+        let helper_func = match func {
```

Comment:
No need for this `match` here, since this function is only called after we've already matched the `func`. You can just delete that parameter from this function.

Reply:
You will need to match on the type though, and we'll need different `is-eq` functions for different types. Your current function works for `int` and `uint` but we'll need to support other types too. For this PR, let's rename the current function `is-eq-int` and call it for both `int` and `uint` types, then support the other types in a future PR. @Acaccia mentioned wanting to work on the comparisons for other types next.